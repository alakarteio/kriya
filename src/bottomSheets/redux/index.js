import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
