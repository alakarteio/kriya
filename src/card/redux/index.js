export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
