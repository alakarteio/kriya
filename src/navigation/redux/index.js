export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
