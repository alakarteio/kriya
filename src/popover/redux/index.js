export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
