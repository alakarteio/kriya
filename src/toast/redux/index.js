import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

