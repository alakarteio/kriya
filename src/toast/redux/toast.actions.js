export const CREATE_TOAST = 'CREATE_TOAST'

const action = type =>
  (title, options) =>
  ({ type: CREATE_TOAST, payload: { title, type, ...options } })

export const create = (title, options) => action()(title, options)
const types = ['success', 'warning', 'error']
types.forEach((type) => { create[type] = action(type) })

export const REMOVE_TOAST = 'REMOVE_TOAST'
export const remove = () => ({ type: REMOVE_TOAST })
