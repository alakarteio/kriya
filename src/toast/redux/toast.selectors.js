export const getCurrentToast = state => state.ui.toast[0]
