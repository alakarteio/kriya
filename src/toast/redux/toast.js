import { CREATE_TOAST, REMOVE_TOAST } from './toast.actions'

export const initState = {}
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case CREATE_TOAST:
      return { ...payload, print: true }
    case REMOVE_TOAST:
      return initState
    default: return state
  }
}
