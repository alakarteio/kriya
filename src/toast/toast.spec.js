/* eslint-disable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
/* eslint-env jest */
import React from 'react'
import snap from 'snap'
import mockComponent from '__mocks__/component'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Toast from './toast'
import { REMOVE_TOAST, remove } from './redux'
import ToastContainer from './toast.container'

jest.mock('../button', () => mockComponent('button'))

const snapshot = props => snap(Toast)({ children: [], ...props })
const createContainer = (store) => {
  return mount(
    <Provider store={store}>
      <ToastContainer />
    </Provider>,
  )
}

describe('common/Toast', () => {
  describe('graphical (JSX)', () => {
    it('should have a default behaviour', snapshot({}))
    it('should add custom className', snapshot({ className: 'custom' }))
    it('should add custom style', snapshot({ style: { backgroundColor: 'red' } }))
    it('should add title', snapshot({ title: 'my awesome title' }))
    it('should add button', snapshot({ button: { title: 'my button' } }))
    it('should add button and onClick', snapshot({ button: { title: 'my button', onClick: () => 'ok' } }))
    it('should add button and onClick and extends name', snapshot({ button: { title: 'my button', onClick: () => 'ok', name: 'TOAST_NAME' } }))
    it('should add custom delay', snapshot({ delay: 1000 }))
    it('should add custom type', snapshot({ type: 'warning' }))
    it('should add custom position', snapshot({ position: 'top' }))
    it('should print toast', snapshot({ print: true }))
  })
  describe('container', () => {
    const snapContainer = (state, props, store) => {
      const component = renderer.create(
        <Provider store={store || createStore(() => ({ ui: { toast: state } }))}>
          <ToastContainer {...props} />
        </Provider>,
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }
    it("should pass position when it's defined in props", () => snapContainer({}, { position: 'top' }))
    it(`should dispatch ${REMOVE_TOAST} when remove is called`, () => {
      const dispatch = jest.fn()
      const store = createStore(() => ({ ui: { toast: { print: true, title: 'toast' } } }))
      store.dispatch = dispatch
      const container = createContainer(store)
      container.find(Toast).props().remove()
      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([remove()])
    })
  })
})

/* eslint-enable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/

