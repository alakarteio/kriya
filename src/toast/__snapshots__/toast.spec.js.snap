// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Toast container should pass position when it's defined in props 1`] = `
<div
  className="toast top default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add button 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
  <div
    className="button"
  >
    {"name":"button","props":{"name":"TOAST_default","primary":true,"flat":true,"title":"my button"}}
  </div>
</div>
`;

exports[`common/Toast graphical (JSX) should add button and onClick 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
  <div
    className="button"
  >
    {"name":"button","props":{"name":"TOAST_default","primary":true,"flat":true,"title":"my button"}}
  </div>
</div>
`;

exports[`common/Toast graphical (JSX) should add button and onClick and extends name 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
  <div
    className="button"
  >
    {"name":"button","props":{"name":"TOAST_NAME","primary":true,"flat":true,"title":"my button"}}
  </div>
</div>
`;

exports[`common/Toast graphical (JSX) should add custom className 1`] = `
<div
  className="toast custom bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add custom delay 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add custom position 1`] = `
<div
  className="toast top default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add custom style 1`] = `
<div
  className="toast bottom default"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add custom type 1`] = `
<div
  className="toast bottom warning"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should add title 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    my awesome title
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should have a default behaviour 1`] = `
<div
  className="toast bottom default"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;

exports[`common/Toast graphical (JSX) should print toast 1`] = `
<div
  className="toast bottom default print"
  style={Object {}}
>
  <span
    className="text"
  >
    
  </span>
</div>
`;
